lib/internal/process/pre_execution.js

```diff
@@ -498,6 +503,48 @@ function initializeClusterIPC() {
   }
 }
 
+function initializePermission() {
+  const experimentalPermission = getOptionValue('--experimental-permission');
+  if (experimentalPermission) {
+    process.emitWarning('Permission is an experimental feature',
+                        'ExperimentalWarning');
+    const { has, deny } = require('internal/process/permission');
+    const warnFlags = [
+      '--allow-child-process',
+      '--allow-worker',
+    ];
+    for (const flag of warnFlags) {
+      if (getOptionValue(flag)) {
+        process.emitWarning(
+          `The flag ${flag} must be used with extreme caution. ` +
+        'It could invalidate the permission model.', 'SecurityWarning');
+      }
+    }
+
+    ObjectDefineProperty(process, 'permission', {
+      __proto__: null,
+      enumerable: true,
+      configurable: false,
+      value: {
+        has,
+        deny,
+      }
+    });
+  } else {
+    const availablePermissionFlags = [
+      '--allow-fs-read',
+      '--allow-fs-write',
+      '--allow-child-process',
+      '--allow-worker',
+    ];
+    for (const flag of availablePermissionFlags) {
```

Comment:
it might be worth using ArrayPrototypeForEach or a standard for loop here, to avoid the iterator protocol